specs/PeriodicBackgoundSyncAndBackgroundSync.md

```diff
@@ -0,0 +1,581 @@
+        // Indicates that the synchronization is a periodic background synchronization.
+        // See [Periodic Background Synchronization](https://developer.mozilla.org/docs/Web/API/Web_Periodic_Background_Synchronization_API)
+        // for more information.
+        PeriodicSync = 1,
+    };
+
+    runtimeclass CoreWebView2ServiceWorkerRegistration
+    {
+        CoreWebView2ServiceWorkerSyncRegistrationManager SyncRegistrationManager { get; };
+    }
+
+    runtimeclass CoreWebView2ServiceWorker
+    {
+        Windows.Foundation.IAsyncOperation<Boolean> DispatchPeriodicSyncEventAsync(String Tag);
+    }
+
+    runtimeclass CoreWebView2ServiceWorkerSyncRegistrationManager
+    {
+        
+        event Windows.Foundation.TypedEventHandler<CoreWebView2ServiceWorkerSyncRegistrationManager, CoreWebView2ServiceWorkerSyncRegisteredEventArgs> BackgroundSyncRegistered;
+
+        event Windows.Foundation.TypedEventHandler<CoreWebView2ServiceWorkerSyncRegistrationManager, CoreWebView2ServiceWorkerSyncRegisteredEventArgs> PeriodicSyncRegistered;
+
+        Windows.Foundation.IAsyncOperation<IVectorView<CoreWebView2ServiceWorkerSyncRegistrationInfo>> GetSyncRegistrationsAsync(CoreWebView2ServiceWorkerSynchronizationKind Kind);
+    }
+
+    runtimeclass CoreWebView2ServiceWorkerSyncRegisteredEventArgs
+    {
+        CoreWebView2ServiceWorkerSyncRegistrationInfo RegistrationInfo { get; };
+    }
+
+    runtimeclass CoreWebView2ServiceWorkerSyncRegistrationInfo
+    {
+        Int64 MinInterval { get; };
+        String Tag { get; };
+    }
+}
+```
+
+
+# Examples
+### C++ Sample
+```cpp
+static constexpr WCHAR c_samplePath[] = L"ScenarioServiceWorkerSyncRegistrationManager.html";
+
+ScenarioServiceWorkerSyncRegistrationManager::ScenarioServiceWorkerSyncRegistrationManager(
+    AppWindow* appWindow)
+    : m_appWindow(appWindow), m_webView(appWindow->GetWebView())
+{
+    m_sampleUri = m_appWindow->GetLocalUri(c_samplePath);
+
+    auto webView2_13 = m_webView.try_query<ICoreWebView2_13>();
+    CHECK_FEATURE_RETURN_EMPTY(webView2_13);
+
+    wil::com_ptr<ICoreWebView2Profile> webView2Profile;
+    CHECK_FAILURE(webView2_13->get_Profile(&webView2Profile));
+    auto webViewProfile3 = webView2Profile.try_query<ICoreWebView2Profile3>();
+    CHECK_FEATURE_RETURN_EMPTY(webViewProfile3);
+    wil::com_ptr<ICoreWebView2ServiceWorkerManager> serviceWorkerManager;
+    CHECK_FAILURE(webViewProfile3->get_ServiceWorkerManager(&serviceWorkerManager));
+
+    if (serviceWorkerManager)
+    {
+        CHECK_FAILURE(serviceWorkerManager->GetServiceWorkerRegistration(
+            m_appWindow->GetLocalUri(L"").c_str(),
+            Callback<ICoreWebView2GetServiceWorkerRegistrationCompletedHandler>(
+                [this](
+                    HRESULT error,
+                    ICoreWebView2ServiceWorkerRegistration* serviceWorkerRegistration)
+                    -> HRESULT
+                {
+                    if (serviceWorkerRegistration)

```

Comment:
You should be checking the HRESULT error and don't need to check if serviceWorkerRegistration is non-nullptr?

Reply:
Added the HRESULT check. And Keep the non nullptr check as we can get null serviceWorkerRegistration.